### PR TITLE
trrntzip: init at 1.1

### PR DESCRIPTION
**Pull request by TheBrainScrambler**
## Description of changes

The goal of Torrentzip is to use standard values when creating zips to create identical files over multiple systems. In other words, when zipping a file you should always get the same zip, no matter what implementation of Torrentzip you're using. trrntzip is an implementation of that.

https://github.com/0-wiz-0/trrntzip

This is a revival of the old trrntzip at https://sourceforge.net/projects/trrntzip, see https://sourceforge.net/p/trrntzip/discussion/457469/thread/d3610ea3b6/ . There hasn't been any response, and besides the new one is on github instead of sourceforge which makes life for us easier.

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by keenanweaver** [comment]
Result of `nixpkgs-review pr 253132` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>trrntzip</li>
  </ul>
</details>

**Review by keenanweaver — APPROVED** [review]

**Review by lasers — COMMENTED** [review]
Hi. Can you do a version bump, please? And I tried this just now... I'm getting this...
```shell
❯ ./trrntzip ~/Desktop/Test2.zip
Could not open log file 'error.log'!
```

File: pkgs/tools/archivers/trrntzip/default.nix
Comment (h7x4) [inline pkgs/tools/archivers/trrntzip/default.nix:50]: Could you add yourself as maintainer?

```suggestion
    platforms = platforms.linux;
    maintainers = with maintainers; [ TheBrainScrambler ];
```

Comment (h7x4) [inline pkgs/tools/archivers/trrntzip/default.nix:50]: This looks like it should support a whole bunch of platforms (https://github.com/0-wiz-0/trrntzip/blob/main/src/platform.h, https://github.com/0-wiz-0/trrntzip/blob/main/src/platform.c). I don't know how difficult it would be to get it working on darwin, but I think it's better to list out the platforms we think upstream supports, and rather mark darwin as broken.

```suggestion
    platforms = with platforms; unix ++ windows;
```

File: pkgs/tools/archivers/trrntzip/default.nix
Comment (h7x4) [inline pkgs/tools/archivers/trrntzip/default.nix:30]: Looks like there's some optionals features involving bzip2: https://github.com/0-wiz-0/trrntzip/blob/29d141d02bb677f593b76cf5fead59ef6056eb3d/src/contrib/zip.c#L1003

Not sure how important it is.

File: pkgs/tools/archivers/trrntzip/default.nix
Comment (h7x4) [inline pkgs/tools/archivers/trrntzip/default.nix:50]:
```suggestion
    platforms = platforms.linux;
    mainProgram = "trrntzip";
```

File: pkgs/top-level/all-packages.nix
Comment (h7x4) [inline pkgs/top-level/all-packages.nix:13930]: I think the package should be moved to `pkgs/by-name/tr/trrntzip/package.nix`

File: pkgs/tools/archivers/trrntzip/default.nix
Comment (TheBrainScrambler) [reply pkgs/tools/archivers/trrntzip/default.nix:50]: I've always thought that `meta.platforms` is for platforms that were tested, and not for those that may work in theory. https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#built-on-platforms

File: pkgs/tools/archivers/trrntzip/default.nix
Comment (TheBrainScrambler) [reply pkgs/tools/archivers/trrntzip/default.nix:30]: I have no idea. I don't see why bzip2 would be needed when this is supposed to be a package for zip files.

File: pkgs/tools/archivers/trrntzip/default.nix
Comment (TheBrainScrambler) [reply pkgs/tools/archivers/trrntzip/default.nix:50]: Shouldn't `meta.mainProgram` only be set when necessary ? In this case it's useless as `trrntzip` is also the name of the package

File: pkgs/top-level/all-packages.nix
Comment (TheBrainScrambler) [reply pkgs/top-level/all-packages.nix:13930]: At the time I wrote this package the `pkgs/by-name` hierarchy didn't exist. But now that I'm modifying the package anyway, why not

File: pkgs/tools/archivers/trrntzip/default.nix
Comment (TheBrainScrambler) [reply pkgs/tools/archivers/trrntzip/default.nix:50]: Lol I seriously forgot to do that ? Thanks !

**Comment by TheBrainScrambler** [comment]
> Hi. Can you do a version bump, please? And I tried this just now... I'm getting this...
> 
> ```shell
> ❯ ./trrntzip ~/Desktop/Test2.zip
> Could not open log file 'error.log'!
> ```

I've bumped to 1.1 . But for me it worked on 1.0 and still works on 1.1 . What exactly is your zip file ? How did you make it ?
I've always tested trrntzip this way:
```
cd /tmp
zip test.zip somefile
trrntzip test.zip
```

File: pkgs/tools/archivers/trrntzip/default.nix
Comment (h7x4) [reply pkgs/tools/archivers/trrntzip/default.nix]: I believe the docs you linked have to do with testing before merge requests. I think the default is to mark supported platforms as either specified upstream, or what we suspect it will run on. I'm on mobile atm, but I'll see if I can find some docs about this later

File: pkgs/tools/archivers/trrntzip/default.nix
Comment (h7x4) [reply pkgs/tools/archivers/trrntzip/default.nix]: Seems to be a supported compression mode. But I don't see it reflected in the CMAKE files. Maybe we can just leave it be

File: pkgs/tools/archivers/trrntzip/default.nix
Comment (h7x4) [reply pkgs/tools/archivers/trrntzip/default.nix]: This used to be the case, but these days we set it for everything

**Comment by lasers** [comment]
Whatever I tried... It's gone. I can trrntzip fine on 1.0 and 1.1 now. Idk.

There are now logs in my current working directory. I can't recall if trrntzip.net does that too... Just a heads up about possible unwanted logging in any future working directory... I understand that it's not something you did. Thank you for this. Cheers.

**Review by lasers — APPROVED** [review]

**Comment by lasers** [comment]
>Could not open log file 'error.log'!

I got it. I tabbed wrong `Test` file (i.e. `trrntzip Test.pdf` instead of  `trrntzip Test.zip`).

**Comment by JohnRTitor** [comment]
Result of `nixpkgs-review pr 253132` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>trrntzip</li>
  </ul>
</details>

**Pull request merged**
